File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        unordered_set<int>map(nums.begin(),nums.end());
        int res=0;

        for(const auto &i: map)
        {
            if(!map.count(i-1))
            {
                int len=1;
                while(map.count(len+i))
                {

                    len++;

                }

                res=max(len,res);
            }
        }

        return res;

        
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        #so basically 100-1 exist karto ka, karat asel tar this cannot be the start

        unique=set(nums)
        maxL=0


        for n in unique:
            if n-1 not in unique:
                length=0
                while n+length in unique:
                    length+=1
                maxL=max(length,maxL)

        return maxL

        